apps/src/labs/lab2.tsx

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressReduxSelectors';
+import {ProgressState} from '@cdo/apps/code-studio/progressRedux';
+import {LevelWithProgress} from '@cdo/apps/types/progressTypes';
+import {getStandaloneProjectId} from '@cdo/apps/labs/projects/utils';
+import ProjectContainer from '@cdo/apps/labs/projects/ProjectContainer';
+import ProgressContainer from '@cdo/apps/labs/progress/ProgressContainer';
+import StandaloneVideo from '@cdo/apps/standaloneVideo/StandaloneVideo';
+import MusicView from '@cdo/apps/music/views/MusicView';
+import {useSelector} from 'react-redux';
+
+const Lab2: React.FunctionComponent = () => {
+  const currentApp = useSelector(
+    (state: {progress: ProgressState}) =>
+      levelsForLessonId(state.progress, state.progress.currentLessonId).find(
+        (level: LevelWithProgress) => level.isCurrentLevel
+      ).app
+  );
+
+  const channelId: string | null =
+    currentApp === 'music' ? getStandaloneProjectId() : null;
+
+  return (
+    <ProjectContainer channelId={channelId || undefined}>
+      <div
```

Comment:
Could we make this code more generic, to make it easier to support future labs? I think my approach would be to store 1) a mapping of app -> React class, as well as 2) a map of app -> can be unmounted (if we do continue supporting hiding app views with visibility). Then we can just iterate through our list of known apps and render the correct list of app views, with the appropriate visibility/mounted logic.

Reply:
Just committed a first pass at this idea.  Could definitely be cleaned up, but curious to hear initial thoughts!